HISTORY.rst

```diff
@@ -1,6 +1,18 @@
 History
 =======
 
+Unreleased
+----------
+
+- When opportunistic_auth is enabled but an Authorization header
+  could not be generated opportunistically, log a warning rather
+  than raising an exception. (If the request results in a 401
+  and we fail to generate an Authorization header in that (no-
+  longer-opportunistic) case, an exception is still raised.)
+
+- Never log Authorization header values to prevent unintentional
+  leaking of credentials.
+
```

Comment:
This is not how this file is used.  Please refer to git history, or look at the previous entries.